### ISSUE TITLE
Regression with "MSL: For 2.1+, don't disable rasterization for vertex writes."

### ISSUE THREAD

**Issue by null77** [KhronosGroup/SPIRV-Cross#1570]
This CL: 1b6b9705e6ae699e05705ab64c393aeb554bc28b

Is causing a regression in ANGLE's Metal back-end implementation (GL-on-Metal) with transform feedback.

Can follow-up with more info and reproduction steps as necessary. @kakashidinho @dj2 fyi

**Comment by null77** [comment]
Some context in http://crbug.com/1159974

**Comment by cdavis5e** [comment]
> Can follow-up with more info and reproduction steps as necessary. @kakashidinho @dj2 fyi

Yes, please do.

**Comment by kakashidinho** [comment]
It looks like newer metal indeed supports vertex shader writing to buffers without disabling rasterization.
ANGLE is assuming it was not the case because it relies on old behaviors of spirv-cross (which always enforced the rasterization to be disabled by converting to void function)

@cdavis5e, I tested compiling a non-void vertex shader with writable buffers. It can compile on macOS 10.15. But the compiler still throws this warning "writable resources in non-void vertex function". Are you sure non-void vertex shader with writable buffers will work properly?

**Comment by cdavis5e** [comment]
> Are you sure non-void vertex shader with writable buffers will work properly?

No. I think it might depend on the GPU family the `MTLDevice` supports. I know it works on my AMD Radeon Pro 460 and my Intel HD Graphics 530, both of which support `Mac2`.
